vessim/__init__.py

```diff
@@ -213,55 +229,58 @@ def forecast(
             2020-01-01 01:50:00    2.8
             Freq: 20T, Name: zone_a, dtype: float64
         """
+        start_time = pd.to_datetime(start_time)
+        end_time = pd.to_datetime(end_time)
         data_src = self._get_forecast_data_source(start_time)
 
         zone_index = self._get_zone_index_from_dataframe(data_src, zone)
 
         # Get data of zone and convert to pd.Series
-        forecast: pd.Series = data_src.iloc[:, zone_index].squeeze()
+        forecast: pd.Series = data_src.iloc[:, zone_index].dropna()
 
         # Resample the data to get the data to specified frequency
         if frequency is not None:
-            forecast = self._resample_to_frequency(
+            frequency = pd.tseries.frequencies.to_offset(frequency)
+            if frequency is None:
+                raise ValueError(f"Frequency '{frequency}' invalid.")
+
+            forecast_in_freq: pd.Series = self._resample_to_frequency(
```

Comment:
These changes do not seem to improve performance for lets-wait-awhile. The problem remains that the dataset is being resampled to match the desired frequency. In case of lets-wait-awhile, the original solution for this was a simple modulo operation filtering out unneeded entries. The function _resample_to_frequency() takes up 70% of execution time. 

Let's discuss this tomorrow!

Reply:
but you do know the frequency of your data (if it's consistent), what's the overhead in doing a single check if it matches the frequency provided by the user?